### PR TITLE
Specify in unix.mli which functions are implemented on Windows

### PR DESCRIPTION
**Pull request by Chris00**
Fixes http://caml.inria.fr/mantis/view.php?id=5253

### CODE REVIEW

**Comment by Chris00** [comment]
Revisiting functions in the `Unix` module also brought up some naive questions.
- The API call [system](https://msdn.microsoft.com/en-us/library/277bwbdz.aspx) documentation says "If the command interpreter is not found, returns 0 and sets errno to ENOENT."  It seems that this behavior is not taken into account by the C function `win_system`.
- `nice` is not implemented.  However, Windows has a [SetPriorityClass](https://msdn.microsoft.com/en-us/library/ms686219%28v=vs.85%29.aspx) which could somewhat provide an alternative.
- For `chown` there are [many alternatives](https://serverfault.com/questions/27332/how-to-take-ownership-of-files-from-the-command-line) for  Windows (I haven't investigated further but they clearly indicate  that a syscall must exist).  Maybe it would make sense to have  "higher level" functions that set ownership for named users (using  [pwd.h](http://pubs.opengroup.org/onlinepubs/000095399/basedefs/pwd.h.html) on  the Unix side)?

File: otherlibs/unix/unix.mli [hunk 1]
```diff
@@ -298,6 +309,8 @@ val in_channel_of_descr : file_descr -> in_channel
    [set_binary_mode_in ic false] if text mode is desired.
    Text mode is supported only if the descriptor refers to a file
    or pipe, but is not supported if it refers to a socket.
+   Under Windows, [set_binary_mode_in] always fails on channels created
```
Comment (gasche) [inline otherlibs/unix/unix.mli]: Any particular reason for using either "On Windows" or "Under Windows"?

File: otherlibs/unix/unix.mli [hunk 2]
```diff
@@ -460,21 +485,27 @@ val chmod : string -> file_perm -> unit
...
 val umask : int -> int
 (** Set the process's file mode creation mask, and return the previous
-    mask. *)
+    mask.
+    On Windows: not implemented. *)
 
 val access : string -> access_permission list -> unit
 (** Check that the process has the given permissions over the named
-   file. Raise [Unix_error] otherwise. *)
+   file. Raise [Unix_error] otherwise.
```
Comment (gasche) [inline otherlibs/unix/unix.mli]: This is an opportunity to use ocamldoc's `@raise` tag.

File: otherlibs/unix/unix.mli [hunk 3]
```diff
@@ -899,20 +962,20 @@ val getlogin : unit -> string
 (** Return the login name of the user executing the process. *)
 
 val getpwnam : string -> passwd_entry
-(** Find an entry in [passwd] with the given name, or raise
-   [Not_found]. *)
+(** Find an entry in [passwd] with the given name, or raise [Not_found].
+   On Windows, always raise [Not_found]. *)
```
Comment (gasche) [inline otherlibs/unix/unix.mli:971]: maybe `@raise` here as well.

**Comment by gasche** [comment]
This is very nice, thank you. I think that the PR would be quickly mergeable, and is probably not a good place to discuss the various issues you raise in your second message. Could you create three mantis tickets for them?

File: otherlibs/unix/unix.mli (hunk 1 above)
Comment (Chris00) [inline otherlibs/unix/unix.mli]: No.  Do you prefer the latter?

Comment (gasche) [inline otherlibs/unix/unix.mli]: I think we would better ask a native speaker here, or maybe try to see what the documentation already uses. I must say that I'm not very convinced by "On windows: not available", I'd rather read "Not available {on,under} Windows".

Comment (Emm) [inline otherlibs/unix/unix.mli]: As non-native English speaker (but who has been speaking English daily for 11 years), I'd say "On Windows"

**Comment by Chris00** [comment]
Thanks for your review.

The mantis tickets [have](http://caml.inria.fr/mantis/view.php?id=7088) [been](http://caml.inria.fr/mantis/view.php?id=7089) [created](http://caml.inria.fr/mantis/view.php?id=7090).

**Comment by Chris00** [comment]
I have not changed (yet) the "On Windows" prefix — I wanted it to stand out without having to read the sentence but I can change that if native English speakers disagree.

**Comment by gasche** [comment]
Merged, thanks!

**Comment by Pireax** [comment]
Wouldn't it make more sense to extract the platform agnostic parts to another module, make the functions more generic and deprecate and then extract the Unix module from the ocaml standard library(slowly and carefully as not to break too many things)? This would incentivise 3rd party libraries to make their libraries also platform agnostic. And if they really need the Unix module they can just add it as a dependency. This would be a big change but I think it would be a good step to make OCaml more accesible and friendlier to use.

**Pull request merged**